forge/lang/expander.rkt

```diff
@@ -759,13 +759,20 @@
   [pwd:PropertyWhereDeclClass
    
     ;; Figure out LHS and RHS (this is ugly, will)
-  #:with imp_lhs (if (eq? (syntax-e #'pwd.constraint-type) 'overconstraint) (format-id stx "~a" #'pwd.prop-name #:source stx) (format-id stx "~a" #'pwd.pred-name #:source stx))
-  #:with imp_rhs (if (eq? (syntax-e #'pwd.constraint-type) 'overconstraint) (format-id stx "~a" #'pwd.pred-name #:source stx) (format-id stx "~a" #'pwd.prop-name #:source stx))
-  #:with test_name (format-id stx "~a-implies-~a" #'imp_lhs #'imp_rhs)
+  ;; #:with imp_lhs (if (eq? (syntax-e #'pwd.constraint-type) 'overconstraint) (format-id stx "~a" #'pwd.prop-name #:source stx) (format-id stx "~a" #'pwd.pred-name #:source stx))
+  ;; #:with imp_rhs (if (eq? (syntax-e #'pwd.constraint-type) 'overconstraint) (format-id stx "~a" #'pwd.pred-name #:source stx) (format-id stx "~a" #'pwd.prop-name #:source stx))
+  ;; #:with test_name (format-id stx "~a-implies-~a" #'imp_lhs #'imp_rhs)
   
    #:with imp_total (if (eq? (syntax-e #'pwd.constraint-type) 'overconstraint)
                         (syntax/loc stx (implies pwd.prop-name pwd.pred-name))  ;;; Overconstraint : Prop => Pred
                         (syntax/loc stx (implies pwd.pred-name pwd.prop-name))) ;;; Underconstraint Pred => Prop
+
+   #:with test_name (let* 
+                      ([imp (syntax->datum #'imp_total)]
+                       [opr (car imp)]
+                       [lhs (format "~a" (car (cdr imp)))]
```

Comment:
Remove the `format`s and use `first` `second` `third` to get the list elements.

Reply:
This could also work:

```
#:do [(match-define (list op lhs rhs) (syntax->list #'imp_total))]
#:with test_name (format-id stx "~a ~a ~a" lhs op rhs)
```

(assuming `racket/match` is required `for-syntax`).

Reply:
Unfortunately, I'm having some trouble with the imports (first, second, third). But hopefully, this update (using car, caddr, etc) and hopefully it is sufficiently succinct.

Reply:
That's probably because you need `(require (for-syntax racket/list))` to make `first`, `second`, and `third` usable. Similarly, if you want to use `match-define`, you would need `(require (for-syntax racket/match))`.

Reply:
Ah, I see. Thanks!